zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/X509AuthenticationUtil.java

```diff
@@ -184,17 +168,16 @@ public static String getClientId(ServerCnxn cnxn, X509TrustManager trustManager)
    */
   private static String matchAndExtractSAN(X509Certificate clientCert)
       throws CertificateParsingException {
-    Integer matchType = Integer.getInteger(SSL_X509_CLIENT_CERT_ID_SAN_MATCH_TYPE);
-    String matchRegex = System.getProperty(SSL_X509_CLIENT_CERT_ID_SAN_MATCH_REGEX, ".*");
-    String extractRegex = System.getProperty(SSL_X509_CLIENT_CERT_ID_SAN_EXTRACT_REGEX, ".*");
+    Integer matchType = X509AuthenticationConfig.getInstance().getClientCertIdSanMatchType();
+    String matchRegex = X509AuthenticationConfig.getInstance().getClientCertIdSanMatchRegex();
+    String extractRegex = X509AuthenticationConfig.getInstance().getClientCertIdSanExtractRegex();
     Integer extractMatcherGroupIndex =
-        Integer.getInteger(SSL_X509_CLIENT_CERT_ID_SAN_EXTRACT_MATCHER_GROUP_INDEX);
+        X509AuthenticationConfig.getInstance().getClientCertIdSanExtractMatcherGroupIndex();
     LOG.info("X509AuthenticationUtil::matchAndExtractSAN(): Using SAN in the client cert "
             + "for client ID! matchType: {}, matchRegex: {}, extractRegex: {}, "
             + "extractMatcherGroupIndex: {}", matchType, matchRegex, extractRegex,
         extractMatcherGroupIndex);
-    if (matchType == null || matchRegex == null || extractRegex == null || matchType < 0
-        || matchType > 8) {
+    if (matchRegex == null || extractRegex == null || matchType < 0 || matchType > 8) {
```

Comment:
Looking at setClientCertIdSanMatchType() we should still check if "matchType == null" right?

Reply:
Good point. I changed the return type for getClientCertIdSanMatchType() from `Integer` to `int`, so we don't need to do the null check now. Please review my change again. Thx.